src/components/PostDetails.tsx

```diff
@@ -65,28 +49,18 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(setComments([...comments, newComment]));
```

Comment:
Instead of directly modifying the state by passing the new array to `setComments`, it's better to create an action in the commentsSlice to add a new comment to the state. This will make the code more maintainable.